R/cartogram_ncont.R

```diff
@@ -47,7 +48,13 @@
 #'plot(nc_utm_carto[,"BIR74"], add =TRUE)
 #'
 #' @references Olson, J. M. (1976). Noncontiguous Area Cartograms. In The Professional Geographer, 28(4), 371-380.
-cartogram_ncont <- function(x, weight, k = 1, inplace = TRUE){
+cartogram_ncont <- function(
+  x,
+  weight,
+  k = 1,
+  inplace = TRUE,
+  n_cpu = parallelly::availableCores()
```

Comment:
I think that in general it is recommended to keep the default core use to 1.

Reply:
I agree

Reply:
Actually, having thought of that, I don't fully agree. In case of highly advanced packages such as `mlr3`, I would agree that default should be 1 core and the user should be able to have full control over which algorithm uses how many cores in the inner and outer loops. With simpler functions that users are more likely to use in interactive mode, I would argue that using maximum cores would be better, as most users probably don't read the docs and don't change the defaults. Say, you normally don't let users of GUI apps choose how many cores to use (e.g. in Photoshop, Final Cut, and many more), these kind of software just try to use as many cores as you have to maximise performance, and this default is probably exactly because otherwise users (who almost never touch defaults) would simply complain that software is slow, or does not use their hardware to its maximum potential, or worse, would switch to the competitor software that does do multicore by default. If we take DuckDB as an example, it uses [80% of RAM by default](https://duckdb.org/2024/07/09/memory-management.html) and all of the CPU cores (which I disagree with and would rather use max less 1 core, reserving it for the OS to prevent freezes).

I am currently working on a new commit where I plan to design 3 choices:
- auto mode - similar to DuckDB's approach it would use max cores less 1
- respect future::plan()
- explicit number of cores, where the function will use future::plan() internally

The open question will remain which of these should be the default. I would say that auto mode is a better design choice, as forcing the users of `cartogram` to use future::plan seems like an overkill.

@Nowosad , @sjewo what do you think?